Hugo Config option for blog URLs: avoid deprecation

olleolleolle opened the pull request:
### **User description**
### Description

This PR avoids a warning emitted by latest Hugo at start of generating the website.


### Motivation and Context

We want to follow current Hugo practices, to have a working website.

    WARN  deprecated: the ":filename" permalink token was deprecated in Hugo 0.144.0 and will be removed in a future release. Use ":contentbasename" instead.

Update: hugo v0.125.4 is running in the GitHub Action. So that doesn't have support for that feature yet. 

Now https://github.com/peaceiris/actions-hugo?tab=readme-ov-file#options CAN configure a newer one.  

Update: I chose to update to latest Hugo in the jobs that required a newer one.

### TODO

- [x] Avoid GetJSON, and follow the error message. Usages: https://github.com/search?q=repo%3ASeleniumHQ%2Fseleniumhq.github.io%20GetJSON&type=code `12:02:55 PM: ERROR deprecated: data.GetJSON was deprecated in Hugo v0.123.0 and subsequently removed. use resources.Get or resources.GetRemote with transform.Unmarshal.` Hints in this help article - https://discourse.gohugo.io/t/how-to-replace-the-data-getjson-with-resources-getremote-or-resources-get/52728/4


### Types of changes

- [x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)

### Checklist

- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.


___

### **PR Type**
Other


___

### **Description**
- Replace deprecated `:filename` with `:contentbasename` in Hugo permalinks


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Hugo Config"] -- "update permalink" --> B["Replace :filename"] -- "with" --> C[":contentbasename"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>hugo.toml</strong><dd><code>Update Hugo permalink configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/hugo.toml

<ul><li>Replace deprecated <code>:filename</code> token with <code>:contentbasename</code> in blog <br>permalinks configuration</ul>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/2439/files#diff-7c7b0a30aca04da823c578a790ae39208bfb7bc0195ccc764f74a3f2a049e467">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

olleolleolle committed:
Hugo Config option for blog URLs: avoid deprecation

    WARN  deprecated: the ":filename" permalink token was deprecated in Hugo 0.144.0 and will be removed in a future release. Use ":contentbasename" instead.

diemol reviewed (commented):
It seems the build is failing.

olleolleolle committed:
Refer to latest Hugo release

olleolleolle commented:
In order to have a Hugo that supports the current way of expression, I set hugo-version to the current release in the CI jobs I could find. cc @diemol

diemol reviewed (commented):
Thank you for this. 

However, there are other places where version 0.125.4 is set. Like the `netlify.toml` file.

olleolleolle commented:
I learned that there were a few more places. In situations like this, I often write a script/update_hugo_version shell script (perhaps in Ruby or something that is convenient for me) that updates all the places.

diemol committed:
Merge branch 'trunk' into use-current-hugo-config-for-filename

diemol commented:
This is the error the Netlify build reports. I am not sure why the GitHub Actions build does not report the same.

```
12:02:50 PM: Start building sites …
12:02:50 PM: hugo v0.148.2-40c3d8233d4b123eff74725e5766fc6272f0a84d+extended linux/amd64 BuildDate=2025-07-27T12:43:24Z VendorInfo=gohugoio
12:02:50 PM: WARN  skipping template file "/opt/build/cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/google/docsy@v0.10.0/layouts/_default/_markup/td-render-heading.html": unrecognized render hook template
12:02:53 PM: WARN  The "gist" shortcode was deprecated in v0.143.0 and will be removed in a future release. See https://gohugo.io/shortcodes/gist for instructions to create a replacement.
12:02:55 PM: WARN  The "twitter", "tweet", and "twitter_simple" shortcodes were deprecated in v0.142.0 and will be removed in a future release. Please use the "x" shortcode instead.
12:02:55 PM: ERROR deprecated: data.GetJSON was deprecated in Hugo v0.123.0 and subsequently removed. use resources.Get or resources.GetRemote with transform.Unmarshal.
```

olleolleolle commented:
@diemol There is a concept of “extended Hugo”, perhaps that has a different set of available commands/a different deprecation schedule. On my local macOS brew install, I got the warning about twitter, but not the error there.

diemol commented:
I think the problem is this one:
```
12:02:55 PM: ERROR deprecated: data.GetJSON was deprecated in Hugo v0.123.0 and subsequently removed. use resources.Get or resources.GetRemote with transform.Unmarshal.
```

Yes, we use extended Hugo, because Docsy requires it.

olleolleolle committed:
Hugo: stop using getJSON

I looked at https://discourse.gohugo.io/t/how-to-replace-the-data-getjson-with-resources-getremote-or-resources-get/52728/4 to learn a little about how to do it.

olleolleolle committed:
Merge branch 'trunk' into use-current-hugo-config-for-filename

olleolleolle commented:
@diemol I attempted to make a change, can you "approve the CI build"?

diemol reviewed (approved):
Thank you!
